lib/primitives/mtx.js

```diff
@@ -1688,13 +1711,58 @@ class CoinSelector {
    */
 
   fund() {
-    // Ensure all preferred inputs first.
-    if (this.inputs.size > 0) {
-      const coins = [];
+    this.resolveInputs();
+    this.resolvePreferred();
```

Comment:
Should we just call these `resolveExisting` and `resolvePreferred`? Try to be consistent with the two groups of inputs